jetty-core/jetty-util/src/test/java/org/eclipse/jetty/util/Utf8AppendableTest.java

```diff
@@ -139,6 +139,30 @@ public void testGermanUmlauts(Class<Utf8Appendable> impl) throws Exception
         assertEquals("\u00FC\u00F6\u00E4", buffer.toString());
     }
 
+    /**
+     * Test of ARABIC LETTER TEH MARBUTA.
+     * (hex code point 0629, decimal code point 1577, hex utf-8 bytes D8 A9)
+     * (bidi note, this is a right to left character as well)
+     */
+    @ParameterizedTest
+    @MethodSource("implementations")
+    public void testArabicTehMarbuta(Class<Utf8Appendable> impl) throws Exception
+    {
+        byte[] bytes = new byte[4];
+        bytes[0] = (byte)0xD8;
+        bytes[1] = (byte)0xA9;
+        bytes[2] = (byte)0xD8;
+        bytes[3] = (byte)0xA9;
+
+        Utf8Appendable buffer = impl.getDeclaredConstructor().newInstance();
+        for (int i = 0; i < bytes.length; i++)
+        {
+            buffer.append(bytes[i]);
+        }
+
+        assertEquals("ةة", buffer.toString());
```

Comment:
This test case seems out of place for this PR.
But I thought I discovered a bug in Utf8Appendable with bidi unicode glyphs.
Turns out it works as expected.

Reply:
I like this test.